test/Test/Util/RawPage.hs

```diff
@@ -27,7 +30,7 @@ import           Test.Tasty.QuickCheck (Property, counterexample)
 toRawPage :: PageLogical -> (RawPage, [RawOverflowPage])
 toRawPage p = (page, overflowPages)
   where
-    bs = serialisePage $ encodePage p
+    Just bs = serialisePage <$> encodePage DiskPage4k p
```

Comment:
This is wrong. We do use `toRawPage` to create overflown pages (with big values), and AFAIU `encodePage` will fail then.

Reply:
I think it's _currently_ correct-in-a-sense but only because currently the QC generator for `PageLogical` guarantees that it is small enough that `encodePage` will work. So it's not true for any possible `PageLogical` (since that type can express larger-than-page values), but it is true for the current QC generator.

Yes, that's not terribly satisfying, and I'm planning to change the `PageLogical` type and QC generator to make this more explicit and satisfactory.

Reply:
This should also be resolved by the next PR that changes the property input types (like PageLogical) so that we don't need partial encoding functions.